imagersite/imager_images/views.py

```diff
@@ -1,16 +1,111 @@
 from django.shortcuts import render
 from imager_images.models import Photo, Album
+from django.contrib.auth.decorators import login_required
+from django.http import HttpResponseRedirect
+from django.core.exceptions import PermissionDenied
+from .forms import *
 
 
+@login_required
 def photo_view(request, photo_id):
+    if request.user != Photo.objects.filter(pk=photo_id).first().owner:
+        raise PermissionDenied
     photo = Photo.objects.filter(id=photo_id).first()
     return render(request, 'photo.html', {'photo': photo})
 
 
+@login_required
 def album_view(request, album_id):
+    if request.user != Album.objects.filter(pk=album_id).first().owner:
+        raise PermissionDenied
     album = Album.objects.filter(id=album_id).first()
     return render(request, 'album.html', {'album': album})
 
 
+@login_required
 def library_view(request):
     return render(request, 'library.html')
+
+
+@login_required
+def add_album_view(request):
+    if request.method == 'POST':
+        form = AlbumForm(request.POST, request.FILES)
+        form.fields['photos'].queryset = Photo.objects.filter(owner=request.user)
+        form.fields['cover'].queryset = Photo.objects.filter(owner=request.user)
+        if form.is_valid():
+            new_album = form.save(commit=False)
+            new_album.owner = request.user
+            new_album.save()
+            form.save_m2m()
+            return HttpResponseRedirect('/images/library')
+        else:
+            return render(request, 'add_album.html', {'form': form.as_p})
```

Comment:
notice that this return value is identical to the one at the end of the next else clause.  Why not just return the once and skip this entirely?